### PR TITLE
Yaml stages

### PR DESCRIPTION
**Pull request by livarcocc**
#### Description

This is a planned change to onboard to yaml stages.

#### Customer Impact

Infra-structure change.

#### Regression?

No.

#### Risk

Low

### CODE REVIEW

File: .vsts-dotnet.yml [hunk 1]
```diff
@@ -8,176 +8,189 @@ trigger:
 # variables:
 #   SignType: real
 #   SkipApplyOptimizationData: false
+stages:
+- stage: build
+  displayName: Build and Test
```
Comment (rainersigwald) [inline .vsts-dotnet.yml]: Nit:
```suggestion
  displayName: Build
```

we don't test in this build. We can think about testing in another stage, now that it won't block publish.

Comment (livarcocc) [reply .vsts-dotnet.yml]: Copy paste from Roslyn. Let me fix this real quick.

**Comment by livarcocc** [comment]
Good catch @rainersigwald. I was missing a property on CIBuild.cmd. Added that. I am re-running the pipeline and will retry the test insertion.

**Comment by riarenas** [comment]
* The default channel for Livar's test branch is tied to the ".Net tools - validation" channel.

  `https://github.com/microsoft/msbuild @ refs/heads/yaml_stages -> .NET Tools - Validation`
  This is what causes the validation channel stage to trigger, and the dev channel stage to be skipped.

* https://github.com/microsoft/msbuild/pull/4613/files#diff-64b68deede22be689a5c5a1dc02a7087R15-R16 configures the assets to be marked as validation assets, so they go into the arcade-validation feed.

* The publish packages to azdo feed step get skipped as for now, all channels publish to an azure blob feed and not an azure devops artifacts feed.

* You can see the actual publishing happening in the "add assets location" step: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=2944334&view=logs&j=841c9107-1508-5744-5f38-ba9581e84f26&t=3d448e79-a526-559e-582a-2d42509f1233&l=96

Everything in that build looks normal

**Comment by rainersigwald** [comment]
> * The default channel for Livar's test branch is tied to the ".Net tools - validation" channel.
>   `https://github.com/microsoft/msbuild @ refs/heads/yaml_stages -> .NET Tools - Validation`
>   This is what causes the validation channel stage to trigger, and the dev channel stage to be skipped.

Can you teach me to fish on this part? Where does this mapping live?

**Comment by riarenas** [comment]
> The default channel for Livar's test branch is tied to the ".Net tools - validation" channel.
https://github.com/microsoft/msbuild @ refs/heads/yaml_stages -> .NET Tools - Validation
This is what causes the validation channel stage to trigger, and the dev channel stage to be skipped.
Can you teach me to fish on this part? Where does this mapping live?

Sure! I just got it using darc

`darc get-default-channels --source-repo msbuild`
```
(410)  https://github.com/microsoft/msbuild @ refs/heads/yaml_stages -> .NET Tools - Validation
(25)   https://github.com/Microsoft/msbuild @ refs/heads/master -> .NET Core 3 Dev
(88)   https://github.com/Microsoft/msbuild @ refs/heads/vs16.0 -> .NET Core 3 Release
```

**Comment by rainersigwald** [comment]
Ah, so a channel must exist and be defined (externally to the job itself) for the publish to work, and we have one defined for this branch and for master, and a stale one we should delete for 16.0? Ok.

**Pull request merged**